### DISCUSSION TITLE
What's the best way to fetch FA2 (NFT) owners?

### DISCUSSION THREAD

**Discussion by Dreitser** [ecadlabs/taquito#1246] [Q&A] [answered]
Hi.
We're a bit stuck trying to find out if a wallet is an owner of a certain FA2 (NFT) using Taquito. Can anyone point us in the right direction?

**Comment by claudebarde** [comment, marked as answer]
In general, FA2 contracts have a bigmap (sometimes called `ledger` or `balances`) whose keys are the users’ addresses together with a token id if that’s relevant to the use case.
You can get the storage of the contract with `await contract.storage()` and check if the user’s address is a key of the bigmap, for example, `await storage.ledger.get(USER_ADDRESS)`. If there is no matching key, the promise returns `undefined`, otherwise, it returns the associated value.
